presto-spark-base/src/main/java/com/facebook/presto/spark/PrestoSparkConfig.java

```diff
@@ -36,6 +36,7 @@
     private String storageBasedBroadcastJoinStorage = "local";
     private DataSize sparkBroadcastJoinMaxMemoryOverride;
     private boolean smileSerializationEnabled = true;
+    private int splitAssignmentBatchSize = 1_000_000;
```

Comment:
Wondering if this batch size is too big?  As you mentioned in the summary, a single HiveSplit can take 1.5kb of memory. With this batch size, we are looking at 1.5gb of memory consumption (worst case), before we serialize and compress the splits. What are the cons of making this smaller to further reduce memory usage?

Reply:
This is a little bit of a tradeoff. A 1_000_000 splits is just a 100 splits per partition (assuming the maximum number of partitions is 10000). Getting to lower values might reduce compression efficiency.